### ISSUE TITLE
"Clear" alternative

### ISSUE THREAD

**Issue by extrowerk** [BerryFarm/berrymuch#44]
The userland doesn't provide the common "clear" utility, or at leas i haven't found it yet. Is it possible to include it in Berrymuch? (I know about Ctrl-L, but the muscle-memory is stronger than me.)
Maybe it is just a matter of compiling the netbsd or whatever clear.c.

**Comment by extrowerk** [comment]
This works as supposed:
```
#include <unistd.h>

void clearScreen()
{
  const char *CLEAR_SCREEN_ANSI = "\e[1;1H\e[2J";
  write(STDOUT_FILENO, CLEAR_SCREEN_ANSI, 12);
}
```

**Comment by berryamin** [comment]
Ho !! really great. Yes for sure we'd like to be included in the distribution.
@vaskas any idea where it may fit ?

**Comment by vaskas** [comment]
As usual there is an easy way and a fun way. An easy one is to create a port with the source code above. Easy, useful, doesn't break anything - would be happy to accept it. The fun way is finishing one of the ncurses ports we have in `ports-wip/`. The `clear` command is just `tput` in disguise. An ambitions goal would be to nail the https://github.com/sabotage-linux/netbsd-curses port and to compile some slightly problematic ports we have (e.g. calcurse or vim which don't like certain unicode characters) with it. And hey, you'll get `clear` for free with that :D

**Comment by extrowerk** [comment]
@vaskas: I have made progress with the netbsd-curses version. Some patches here and there, some stuff missing from the POSIX headers, etc. It builds fine, but completely untested.

**Comment by vaskas** [comment]
@extrowerk well the one in ports-wip doesn't build so clearly (pun intended) your version is better. Feel free to override the wip port. A great test case to confirm it working is to compile sc-im against it (also in ports-wip). sc-im is a pretty advanced curses app and runs glitchy on the bb10 ncurses.

**Comment by berryamin** [comment]
@extrowerk yes please give us the opportunity to test . This is already a great contribution to the project !

**Comment by extrowerk** [comment]
Yes, i can share the sources, but that will require me to clean things up first. Will fork the original sources and push my changes into a branch. Do not expect big or smart solutions, however

**Comment by extrowerk** [comment]
Here you go: https://github.com/extrowerk/netbsd-curses/commit/16d9957fa7c28e87cbd07b505bafd5cc617485a0

**Comment by vaskas** [comment]
This is super awesome @extrowerk! Are you planning to incorporate it into a port? Can I help?

**Comment by extrowerk** [comment]
QNX sources were available in 2009. I looked into the source tree and was surprised to see not just the kernel sources but the source code of the userland and some libs too.

Sadly it doesn't have "clear" utility, but it contains sources for ncurses. We can maybe get some inspiration from it.
https://fsck.technology/software/QNX/QNX%206/QNX%20Momentics%20Development%20Suite%206.3.2/Qnx%20Sources%2029%2002%202009.rar

**Comment by karawitan** [comment]
clear has been added to ports/contrib
